app/scripts/controllers/rif/rns/resolver/index.js

```diff
@@ -47,4 +63,187 @@ export default class RnsResolver extends RnsJsDelegate {
+   * @param resolverAddress Address of the new resolver to be setted
+   * @returns {Promise<unknown>}
+   */
+  setResolver(domainName, resolverAddress) {
+    return new Promise((resolve, reject) => {
+      this.send(this.rnsContractInstance, 'setResolver', [namehash.hash(domainName), resolverAddress])
+        .then(result => {
+        console.debug('setResolver success', result);
+        resolve(result);
+      }).catch(error => {
+        console.debug('Error when trying to set resolver', error);
+        reject(error);
+      });
+    });
+  }
+
+  /**
+   * Returns an array of chain addresses for a given domain
+   * @param domainName DomainName with the .rsk extension
+   * @returns {Promise<unknown>}
+   */
+  getChainAddressForResolvers (domainName) {
+    return new Promise((resolve, reject) => {
+      const addrChangedEvent = this.multiChainresolverContractInstance.AddrChanged({node: namehash.hash(domainName)}, {fromBlock: 0, toBlock: 'latest'});
+      const chainAddrChangedEvent = this.multiChainresolverContractInstance.ChainAddrChanged({node: namehash.hash(domainName)}, {fromBlock: 0, toBlock: 'latest'});
+      const arrChains = [];
+      const errorLogs = [];
+      addrChangedEvent.get(function(error, result){
+        if (error) {
+          console.debug('Error when trying to get addrChangedEvent for resolver', error);
+          errorLogs.push(error);
+        }
+        result.forEach(event => {
+          const chainAddrToPush = new ChainAddress(ChainId.RSK, event.args.addr)
+          arrChains[0] = chainAddrToPush;
+        });
+        console.debug('getChainAddressForResolvers success', arrChains);
+      });
+      chainAddrChangedEvent.get(function(error, result){
+        if (error) {
+          console.debug('Error when trying to get chainAddrChangedEvent for resolver', error);
+          errorLogs.push(error);
+          reject(errorLogs);
+        }
+        result.forEach(event => {
+          const chainAddrToPush = new ChainAddress(event.args.chain, event.args.addr)
+          const index = arrChains.findIndex((e) => e.chain === chainAddrToPush.chain);
+          if (index === -1) {
+            arrChains.push(chainAddrToPush);
+          } else {
+            arrChains[index] = chainAddrToPush;
+          }
+        });
+        console.debug('getChainAddressForResolvers success', arrChains);
+        resolve(arrChains);
+      });
+    });
+  }
+
+  /**
+   * Calls the contract and sets a new chain address calling the setChainAddr function in the contract of multichainresolver
+   * @param domainName DomainName with the .rsk extension
+   * @param chain
+   * @param chainAddress
+   * @returns {Promise<unknown>}
+   */
+  setChainAddressForResolver (domainName, chain, chainAddress) {
+    return new Promise((resolve, reject) => {
+      this.send(this.multiChainresolverContractInstance, 'setChainAddr', [namehash.hash(domainName), chain, chainAddress])
```

Comment:
Same thing here about the listener if you want to control the action after the transaction.